src/thor/dao/release_dao.py

```diff
@@ -0,0 +1,179 @@
+import os
+import sqlalchemy as sa
+from . import config
+from models.models import Release
+
+from sqlalchemy.orm import sessionmaker
+from contextlib import contextmanager
+
+import logging
+
+from contextlib import contextmanager
+
+# Implements CRUD functions on the database.
+
+engine = sa.create_engine(config.RELEASE_DATABASE_URL)
+Session = sessionmaker(bind=engine)
+
+logging.basicConfig(level=os.environ.get("LOGLEVEL", "INFO"))
+logging.getLogger("sqlalchemy.engine").setLevel(logging.INFO)
+l = logging.getLogger(__name__)
+
+
+@contextmanager
+def session_scope():
+    """ Provides transactional scope around a series of operations. 
+    In other words, handles the messiness of sessions and commits, 
+    including rollback, so other modules don't have to. """
+
+    session = Session()
+    try:
+        yield session
+        session.commit()
+    except:
+        session.rollback()
+        raise
+    finally:
+        session.close()
+
+
+def manCreateRelease(id, version, result):
+    """ Given int ID, string version, and string result, 
+    creates a Release object, and inserts it into the database controlled
+    by the currently active session. 
+    Assumes that the id given is unique. """
+
+    with session_scope() as session:
+      try:
+        if id in getRKeys():
+            raise Exception(
+                "That keyvalue (" + str(id) + ") is already in the database. "
+            )
+        currentRelease = Release(id=id, version=version, result=result)
+      except Exception as e:

```

Comment:
I'd prefer if we don't have this as a catch all Exception. It would be better to catch specific exception types we except, and have this blow up otherwise so we can troubleshoot a bit easier if something breaks.